### PR TITLE
Add a second managed data source for the same `connectorProvider`

### PR DESCRIPTION
**Pull request by spolu**
Allow the creation of an extra data source for the same `connectorProvider`.

Example usage consists in appending to the data source page URL the following query parameters:
```
https://dust.tt/w/<WORKSPACE_ID>/ds?setupWithSuffixConnector=notion&setupWithSuffixSuffix=personal
```

This adds an extra Notion button that will trigger the creation of a new notion managed data source with name `managed-notion-personal` (suffix added added to the data source name)

I successfully connected 2 notion data sources to 2 different notion spaces on my machine. Everything should work smoothly for Slack, Drive, and GitHub

### CODE REVIEW

File: front/pages/w/[wId]/ds/index.tsx [hunk 1]
```diff
@@ -124,24 +127,32 @@ export const getServerSideProps: GetServerSideProps<{
...
+    connector: ConnectorType | null;
+    fetchConnectorError: boolean;
+  }[] = await Promise.all(
     managedDataSources.map(async (mds) => {
-      if (!mds.connectorId) {
+      if (!mds.connectorId || !mds.connectorProvider) {
         throw new Error(
           // Should never happen, but we need to make eslint happy
-          "Unexpected empty connectorId for managed data sources"
+          "Unexpected empty `connectorId or `connectorProvider` for managed data sources"
```
Comment (fontanierh) [inline front/pages/w/[wId]/ds/index.tsx:141]:
```suggestion
          "Unexpected empty `connectorId` or `connectorProvider` for managed data sources"
```

**Review by fontanierh — DISMISSED** [review]
LGTM, I am surprised that it didn't break more things !

**Comment by spolu** [comment]
@fontanierh to be perfectly transparent I didn't test Chat with this but plan to do so as soon as this is pushed to production.

I will also test having 2 slack (but from the same slack team because I have only one) in dev before merging (and check connectors is happy)

**Comment by spolu** [comment]
But really there is no reason that it wouldn't be happy at this stage given that it works well with Notion

**Comment by spolu** [comment]
Screenshot from 2023-07-03 15-59-58

**Review by fontanierh — APPROVED** [review]
LGTM

**Comment by spolu** [comment]
updated: [💬 Runbook: Slack connector](https://www.notion.so/dust-tt/Runbook-Slack-connector-e59f50c458a64a9ea64f4fd17dacaffb?pvs=4) and [📚 Runbook: Notion connector](https://www.notion.so/dust-tt/Runbook-Notion-connector-b3b082e3f8a94ad9b4d42dfa89748582?pvs=4)

**Pull request merged**